packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { VideoStreamOptions } from '@internal/react-components';
+import { AudioDeviceInfo, PermissionConstraints, VideoDeviceInfo, Call } from '@azure/communication-calling';
+import { RemoteParticipantState, VideoStreamRendererViewState } from '@internal/calling-stateful-client';
+import { CallAdapterState, CallAdapter } from '../../../../../src';
+import { TestCallingState, TestRemoteParticipant } from '../../TestCallingState';
+
+/**
+ * Mock class that implements CallAdapter interface for UI snapshot tests
+ */
+export class MockCallAdapter implements CallAdapter {
+  constructor(testState?: TestCallingState) {
+    this.state = defaultCallAdapterState;
+
+    if (!testState) {
+      return;
+    }
+
+    if (this.state.call) {
+      if (testState.remoteParticipants) {
+        const remoteParticipants = convertTestParticipantsToCallAdapterStateParticipants(testState.remoteParticipants);
+        Object.values(remoteParticipants).forEach((participant) => addMockVideo(participant));
+        this.state.call.remoteParticipants = remoteParticipants;
+      }
+    }
+  }
+
+  state: CallAdapterState;
+
+  addParticipant(participantKey: string, participantState: RemoteParticipantState): void {
+    if (!this.state.call) {
+      return;
+    }
+    this.state.call.remoteParticipants[participantKey] = participantState;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any
```

Comment:
Please drop these eslint directives.

Since none of the arguments are used, you can satisfy the interface with empty arg functions:

```ts
  onStateChange(): void {
    throw Error('Handlers not implemented');
  }
```

I'd also recommend throwing an error from all the handler implementations. That way, if we mistakenly call handlers in the test, the test fails instead of passing silently.

Reply:
Sure. But I actually can't throw an error for onStateChange but I throw Exceptions for the rest and removed the eslint directives